ui/widgets/nuxeo-directory-suggestion.js

```diff
@@ -260,15 +273,26 @@ import { escapeHTML } from './nuxeo-selectivity.js';
       );
     }
 
+    _getItemKey(item) {
+      return `${this.directoryName}:${this.idFunction(item)}`;
+    }
+
     _selectionFormatter(entry) {
       return escapeHTML(entry.absoluteLabel || entry.displayLabel);
     }
 
+    _resultFormatter(item) {
+      const key = this._getItemKey(item);
+      if (!directoryCache[key]) {
+        directoryCache[key] = item;
```

Comment:
This caches all the results that were display to the user on the dropdown. We could be using the selection formatter to only cache those that were actually selected.